packages/core/src/components/forms/numericInput.tsx

```diff
@@ -479,15 +479,11 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
 
         this.shouldSelectAfterUpdate = false;
         this.setState({ value: nextValue });
-        this.invokeOnValueChangeCallback(nextValue);
+        this.invokeValueCallback(nextValue, this.props.onValueChange);
     }
 
-    private invokeOnValueChangeCallback(value: string) {
-        Utils.safeInvoke(this.props.onValueChange, +value, value);
-    }
-
-    private invokeOnButtonClickCallback(value: string) {
-        Utils.safeInvoke(this.props.onButtonClick, +value, value);
+    private invokeValueCallback(value: string, callback: (valueAsNumber: number, valueAsString: string) => void) {
```

Comment:
Ah, finally got it. Say you define only one of the callbacks:

`<NumericInput onButtonClick={...} />` or
`<NumericInput onValueChange={...} />`

If you define a _default_ callback, `Utils.safeInvoke` will never receive an undefined function for whichever callback is undefined; it'll simply re-invoke the default function in `handle[*]ButtonClick` and `incrementValue`. Given that, I think we should go with the explicit type definition implemented above. ^